CMakeLists.txt

```diff
@@ -56,3 +52,12 @@ target_include_directories(${PROJECT_NAME} PUBLIC
 target_link_libraries(${PROJECT_NAME} PUBLIC
   ${SDL2_LIBRARY}
 )
+
+if(MSVC)
+  target_compile_options(${PROJECT_NAME} PRIVATE /W4)# /WX)
+else()
+  target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -pedantic -Werror)
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -flto -fuse-linker-plugin")
```

Comment:
These could probably also be converted into target_compile_options.

Also I could probably add their MSVC equivalents to the other side of the if.